canonical/knative-operators
# add juju config to use custom images for knative serving

This adds to knative-serving:
* the `custom_images` config to to the charm, allowing users to provide overrides to the default images so they can use their own (for example, if they work in an air-gapped environment)
* `DEFAULT_IMAGES` defined in the charm that allow the charm to have an opinion on what images should be used if none are specified
* tests for the above features


### Manual test instructions

```
cd charms/knative-serving
charmcraft pack
juju deploy knative-operator --trust --channel latest/edge
juju deploy ./knative-serving_ubuntu-20.04-amd64.charm --trust --config namespace=knative-serving
kubectl get pod -n knative-serving  # observe the knative serving pods
juju config knative-serving custom_images="{activator: not-a-real-image}"
kubectl get pod -n knative-serving  # observe how a new activator pod is started.  This will not deploy successfully as the image above is not real
juju config knative-serving custom_images=""  # reset back to defaults
kubectl get pod -n knative-serving  # observe the knative serving pods
```
